pkg/policy/cleanup.go

```diff
@@ -20,62 +20,88 @@ func (pc *PolicyController) cleanUpPolicyViolation(pResponse engine.PolicyRespon
 	//    - recursively get owner by queries the api server for owner information of the resource
 
 	// there can be multiple violations as a resource can have multiple owners
-	pvs, err := getPv(pc.pvLister, pc.client, pResponse.Policy, pResponse.Resource.Kind, pResponse.Resource.Namespace, pResponse.Resource.Name)
+	pvs, err := getPVs(pc.pvLister, pc.nspvLister, pc.client, pResponse.Policy, pResponse.Resource.Kind, pResponse.Resource.Namespace, pResponse.Resource.Name)
```

Comment:
As we have the resource.Namespace we can determine if its namespaced or cluster-wide resource, and then we know which violation to lookup.
I would suggest having separate functions for simplicity and we can avoid runtime type checks.
As per the current changes we are using one function to handle both types, keep it simple and easier to understand. 

Or if you want to improve the design u have an interface which implements handlers for cluster and namespace violation.

Reply:
Change to use separate functions to handle cleanup for cluster and namespaced violations, as runtime type checks may be less efficient.